### DISCUSSION TITLE
error when generating lock file if Pipfile contains incomplete dependency version

### DISCUSSION THREAD

**Discussion by vishal180618** [pypa/pipenv#5426] [Show and tell]
`pipenv.patched.notpip._vendor.resolvelib.resolvers.InconsistentCandidate: Provided candidate LinkCandidate('https://files.pythonhosted.org/packages/c3/06/e0300cb5f9b5ff9b6d0accdd3536c01bd2300f8154781455914752ab8903/marshmallow-3.18.0-py3-none-any.whl#sha256=35e02a3a06899c9119b785c12a22f4cda361745d66a71ab691fd7610202ae104 (from https://pypi.org/simple/marshmallow/) (requires-python:>=3.7)') does not satisfy SpecifierRequirement('marshmallow~=3.3'), SpecifierRequirement('marshmallow~=3.3.0')`

Pipfile content
```
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
marshmallow = "~=3.3.0"
abcd = {ref = "1.0.8", git = "https://****@github.com/org/abcd-Utilities"}

[dev-packages]

[requires]
python_version = "3.7"
```
abcd contains marshmallow `~=3.3` in setup.py

**Comment by matteius** [comment]
Please see my response here:  https://github.com/pypa/pipenv/issues/5428#issuecomment-1343981813
